backend/src/middleware/firebase-auth/index.js

```diff
@@ -50,3 +50,24 @@ export const checkUser = (req, res, next) => {
     return handleApiError(res, new AuthenticationError());
   }
 };
+
+// Check the token is teacher role 
+export const checkTeacher = (req, res, next) => {
+  getAuthToken(req, res, async () => {
+    try {
+      const { authToken } = req;
+      const userInfo = await admin
+        .auth()
+        .verifyIdToken(authToken);
+
+      if (userInfo.teacher === true) {
+        req.authId = userInfo.uid;
+        return next();
+      }
+
+      return handleApiError(res, new AuthenticationError());
+    } catch (error) {
+      return handleApiError(res, error);
+    }
+  });
```

Comment:
I think this can be improved a little bit. Make the assumption that when using the `checkTeacher` middleware it will always be called after the `checkToken` middleware which attaches user information to the request.

So expand the `checkTeacher` function to attach the authenticated users claim to the request (under `req.userClaims` or `req.userRoles` would be good) so then in the `checkTeacher ` function we just check the req if the role is present and throw an error if not (same pattern as the `checkUser` function). This will save us a another call to firebase speeding up requests.